fix: Improve Yearly Reading Goal modal for better usability and clarity

akramcodez opened the pull request:
Fixes #11324

## Description
This PR implements the design improvements suggested in issue #11324 to enhance the "Yearly Reading Goal" modal's usability and visual clarity.

## Changes Made

### Layout Improvements
- **Vertical Structure**: Reorganized modal with clean top-to-bottom flow
- **Inline Question**: Moved "Learn More" link to same line as question text
- **Right-aligned Controls**: Positioned input field and submit button on the right side
- **Better Edit Flow**: Instruction text now appears before input when editing existing goals

### Typography & Styling
- **Larger Font Sizes**: Increased input and button text from 12px to 14px for better readability
- **Consistent Spacing**: Added proper gaps and padding throughout the modal
- **Enhanced Focus States**: Improved input field focus indicators

### Responsive Design
- **Mobile Optimization**: Maintains horizontal layout on small screens with appropriate sizing
- **Consistent Experience**: Same interaction patterns across all device sizes

## Screenshots

### Before
<img width="1962" height="586" alt="image" src="https://github.com/user-attachments/assets/f45e479b-75bd-4861-a5fc-fd868fa29810" />
<img width="1918" height="445" alt="image" src="https://github.com/user-attachments/assets/eddd320a-baed-45b4-a200-170948e95958" />
<img width="595" height="1218" alt="image" src="https://github.com/user-attachments/assets/44237842-d8eb-4d05-b266-e30adfcc90a0" />


### After
<img width="1901" height="490" alt="image" src="https://github.com/user-attachments/assets/5bcb3c0a-be7f-4b08-b79f-b5b2ef1f8131" />
<img width="1560" height="483" alt="image" src="https://github.com/user-attachments/assets/5a032ed8-ce5d-4728-b137-0305885866c6" />
<img width="567" height="1219" alt="image" src="https://github.com/user-attachments/assets/79931e60-c37a-45fb-8ae3-1ef53158b2eb" />

jimchamp reviewed (changes requested):
Sorry for the delay, @akramcodez.

I cannot merge this until the overflow issue is fixed for smaller screen sizes (mentioned [here](https://github.com/internetarchive/openlibrary/pull/11343#pullrequestreview-3333480325)).

I still don't understand how the focus indicators were improved.  Can you tell me?

jimchamp commented on static/css/components/reading-goal.less:
```diff
@@ -1,17 +1,128 @@
+  }
+}
+
+@media only screen and (max-width: 480px) {
```
We can't have hard-coded breakpoints.  Please use one of [these three](https://github.com/internetarchive/openlibrary/blob/44b9fa1c99609991230193ec281e45efe4647db4/static/css/less/breakpoints.less#L3-L5)

jimchamp reviewed (commented):
I scrolled right past your "Before" picture in [this post](https://github.com/internetarchive/openlibrary/pull/11343#issuecomment-3400210097) -- sorry about that!  The focus indicators appear blue in production today in FF, but not in Chrome (which I didn't know).

The x-overflow issue is happening in production today, and is unrelated to these changes (again -- so sorry!).

It might make sense to rollback the previous commit if the changes were made solely for the purpose of fixing the overflow.

akramcodez commented:
@jimchamp that’s completely okay, I learned a lot from this issue. Instead of reverting that commit, I just removed the small-screen CSS and also adjusted the input and submit button sizes.

<img width="2755" height="1508" alt="image" src="https://github.com/user-attachments/assets/9d2bca73-bb29-4c82-bf9f-eb7741cf64e3" /> 
<img width="850" height="1432" alt="image" src="https://github.com/user-attachments/assets/0d324491-8f70-41c7-a766-fed8717278bb" />

jimchamp reviewed (approved):
Thanks for your patience, @akramcodez!
